rtl/eclrtl/rtldynfield.cpp

```diff
@@ -510,3 +510,481 @@ extern ECLRTL_API int getFieldNum(const char *fieldName, IOutputMetaData &  meta
+                            throwUnexpected();
+                        case type_record:
+                        case type_table:
+                        {
+                            const RtlRecord *subDest = destRecInfo.queryNested(idx);
+                            const RtlRecord *subSrc = sourceRecInfo.queryNested(info.matchIdx);
+                            info.subTrans = new GeneralRecordTranslator(*subDest, *subSrc);
+                            if (!info.subTrans->needsTranslate())
+                            {
+                                // Child does not require translation, but check linkcount mode matches too!
+                                if (type->isLinkCounted())
+                                    if (sourceType->isLinkCounted())
+                                        info.matchType = match_link;
+                                    else
+                                        info.matchType = match_recurse;
+                                else
+                                    if (sourceType->isLinkCounted())
+                                        info.matchType = match_recurse;
+                                    else
+                                        info.matchType = match_perfect;
+                                if (info.matchType != match_recurse)
+                                {
+                                    delete info.subTrans;
+                                    info.subTrans = nullptr;
+                                }
+                            }
+                            else if (info.subTrans->canTranslate())
+                                info.matchType = match_recurse;
+                            else
+                                info.matchType = match_fail;
+                            break;
+                        }
+                        default:
+                            info.matchType = match_fail;
+                            break;
+                        }
+                    }
+                }
+                else if (type->fieldType==sourceType->fieldType)
+                {
+                    if (type->length==sourceType->length)
+                    {
+                        info.matchType = match_perfect;
+                    }
+                    else if (type->isFixedSize() && !type->isNumeric())
+                    {
+                        info.matchType = match_typecast;
+                        if (type->length < sourceType->length)
+                        {
+                            if (type->canTruncate())
+                                info.matchType = match_truncate;
+                        }
+                        else
+                        {
+                            if (type->canExtend(info.fillChar))
+                                info.matchType = match_extend;
+                        }
+                    }
+                }
+                else
```

Comment:
I thought about little-endian ints and decided to ignore for now - mainly I was not sure whether swapInt meant 'big endian' or 'different endian to host machine' and what would happen if we ran on a bigendian machine